test/unit/year_range_test.rb

```diff
@@ -2,16 +2,50 @@
 
 module SmartAnswer
   class YearRangeTest < ActiveSupport::TestCase
-    setup do
-      @year_range = YearRange.new(begins_on: Date.parse('2000-02-01'))
+    context 'beginning on 7th June, 2001' do
```

Comment:
I wonder whether it might be a bit clearer if the context was "beginning and ending in non leap years", or "beginning in the year following a leap year"?